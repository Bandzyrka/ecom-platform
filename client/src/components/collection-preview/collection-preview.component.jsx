import React from 'react';
import CollectionItem from '../collection-item/collection-item.components'
import {CollectionPreviewContainer, TitleContainer, PreviewContainer } from './collection-preview.styles'
import { useNavigate } from "react-router-dom";

export const CollectionPreview = ({title, items, routeName,}) => {
    console.log(title, items)
    let navigate = useNavigate();
    return(
    <CollectionPreviewContainer>
        <TitleContainer onClick={() => navigate(`${routeName}`)}>
            {title.toUpperCase()}
        </TitleContainer>
        <PreviewContainer>
            {
            items
                .filter((item, index )=> index < 4)
                .map(item => (
                    <CollectionItem key={item.id} item={item} />
             ))}
        </PreviewContainer>
    </CollectionPreviewContainer>
    );
}
export default CollectionPreview;