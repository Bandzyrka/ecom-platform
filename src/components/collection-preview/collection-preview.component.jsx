import React from 'react';
import CollectionItem from '../collection-item/collection-item.components'
import {CollectionPreviewContainer, CollectionPreviewTitle, PreviewContainer } from './collection-preview.styles.jsx'

const CollectionPreview = ({title, items}) => (
    <CollectionPreviewContainer>
        <CollectionPreviewTitle> {title.toUpperCase()}</CollectionPreviewTitle>
        <PreviewContainer>
            {
            items
                .filter((item, index )=> index < 4)
                .map(item => (
                    <CollectionItem key={item.id} item={item} />
             ))}
        </PreviewContainer>
    </CollectionPreviewContainer>
);
export default CollectionPreview;