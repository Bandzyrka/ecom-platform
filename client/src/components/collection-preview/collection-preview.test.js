import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './collection-preview.component';

describe('CollectionPreview component', () => {
  let wrapper;

 
  const mockRouteName = 'hats';

  beforeEach(() => {
    
    const mockProps = {

      routeName: mockRouteName,
      title: 'hats',
      items: []
    };
    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  it('should render CollectionPreview component', () => {
    expect(wrapper).toMatchSnapshot();
  });
}