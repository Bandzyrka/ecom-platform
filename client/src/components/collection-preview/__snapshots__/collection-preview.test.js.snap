// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render CollectionPreview component 1`] = `ShallowWrapper {}`;

exports[`CollectionPreview component should render collection preview 1`] = `ShallowWrapper {}`;
